src/binary/ibc-setup/commands/init.spec.ts

```diff
@@ -0,0 +1,86 @@
+import fs from 'fs';
+
+import test from 'ava';
+import axios from 'axios';
+import sinon from 'sinon';
+
+import { Options, run } from './init';
+
+const consoleLog = sinon.stub(console, 'log');
+const fsExistSync = sinon.stub(fs, 'existsSync');
+const fsMkdirSync = sinon.stub(fs, 'mkdirSync');
+const axiosGet = sinon.stub(axios, 'get');
+const fsReadFileSync = sinon.stub(fs, 'readFileSync');
+const fsWriteFileSync = sinon.stub(fs, 'writeFileSync');
+sinon.replace(
+  fs,
+  'lstatSync',
+  sinon.fake.returns({
+    isDirectory: () => true,
+    isFile: () => true,
+  })
+);
+
+test.beforeEach(() => {
+  sinon.reset();
+});
+
+test('read existing registry.yaml', async (t) => {
+  const options: Options = {
+    home: '/home/user',
+    src: 'AAA',
+    dest: 'BBB',
+  };
+
+  fsExistSync.onCall(0).returns(true).onCall(1).returns(true);
+
+  const registryPath = `${options.home}/registry.yaml`;
+  const registryYaml = `
+  version: 1
+  `;
+
+  axiosGet.resolves({
+    data: registryYaml,
+  });
+  fsReadFileSync.returns(registryYaml);
+  fsWriteFileSync.returns(undefined);
+
+  await run(options);
+
+  t.assert(fsMkdirSync.notCalled);
+  t.assert(axiosGet.notCalled);
+  t.assert(fsReadFileSync.calledOnceWith(registryPath));
+  t.assert(fsWriteFileSync.notCalled);
+  t.assert(consoleLog.calledWith({ version: 1 }));
+});
+
+test('initialize home directory and pull registry.yaml from remote', async (t) => {
```

Comment:
Wow, nice work here stubbing this.
I would just made a temp dir with some files and ran it.

Of course, that approach can never actually test most error cases, only happy path. It is definitely nice to show that this is testable.